great_expectations/data_context/store/gx_cloud_store_backend.py

```diff
@@ -102,13 +102,13 @@ class GXCloudStoreBackend(StoreBackend, metaclass=ABCMeta):
     )
 
     _ENDPOINT_VERSION_LOOKUP: dict[str, EndpointVersion] = {
-        GXCloudRESTResource.CHECKPOINT: EndpointVersion.V0,
+        GXCloudRESTResource.CHECKPOINT: EndpointVersion.V1,
         GXCloudRESTResource.DATASOURCE: EndpointVersion.V1,
         GXCloudRESTResource.DATA_ASSET: EndpointVersion.V1,
         GXCloudRESTResource.DATA_CONTEXT: EndpointVersion.V1,
         GXCloudRESTResource.DATA_CONTEXT_VARIABLES: EndpointVersion.V1,
         GXCloudRESTResource.EXPECTATION_SUITE: EndpointVersion.V1,
-        GXCloudRESTResource.VALIDATION_DEFINITION: EndpointVersion.V0,
+        GXCloudRESTResource.VALIDATION_DEFINITION: EndpointVersion.V1,
         GXCloudRESTResource.VALIDATION_RESULT: EndpointVersion.V0,
```

Comment:
Let's also turn on `ValidationResult`? I can issue a follow-up PR that removes this dictionary since we won't need it after everything has been set to V1